anchore_engine/services/catalog/api/controllers/archives.py

```diff
@@ -350,32 +356,58 @@ def archive_image_analysis(imageReferences):
 
         results = []
 
-        for digest in imageReferences:
-            try:
-                # Do synchronous part to start the state transition
-                task = ArchiveImageTask(
-                    account=ApiRequestContextProxy.namespace(), image_digest=digest
-                )
-                result_status, result_detail = task.run()
-                results.append(
-                    {
-                        "digest": task.image_digest,
-                        "status": result_status,
-                        "detail": result_detail,
-                    }
-                )
-            except Exception as ex:
-                logger.exception(
-                    "Unexpected an uncaught exception from the archive task execution"
-                )
-                results.append({"digest": digest, "status": "error", "detail": str(ex)})
-
+        logger.info("Images to archive: %d", len(imageReferences))
+
+        with concurrent.futures.ThreadPoolExecutor(
```

Comment:
I understand the reason for this change and it makes sense in terms of implementation, but it has a side-effect of enabling users to put quite a bit of load on the system in ways the operator might not expect. So, I'd like to see this stay serial so that user API requests cannot cause thread growth in unexpected ways for the admin. The thread workers should be limited to controlled async processes. If a user wants to speed this up they can send parallel requests instead of a single request and those will get queued/throttled by the api thread logic appropriately.